### ISSUE TITLE
Launch for Camel routes should use * and not the yaml file name

### ISSUE THREAD

**Issue by lhein** [KaotoIO/vscode-kaoto#556]
If you want to prototype a route which contains Java files then the current Launch command is not sufficient. The current command is only useful if the user just has a single yaml file that they want to run. I would like to see the launch use the `camel run * --dev` instead so all the Java files and possible other files are taken into consideration too.

**Comment by lhein** [comment]
It doesn't help to store the java class under `src/main/java`.

**Comment by lordrip** [comment]
Couldn't we have an additional button to say "Run container folder"? This way we could have both scenarios covered

![image](https://github.com/KaotoIO/vscode-kaoto/assets/16512618/0fa701ea-2e38-4070-a1af-86fd32fd14cc)

**Comment by apupier** [comment]
yes, it is possible to have a dropdown with this quick button menu action on editors.

I think we should keep 2 distincts buttons between run and debug especially if we are providing 2 entries to run single file and run all files.

**Comment by apupier** [comment]
> Maybe another thing to consider would be if it's possible to run the file in the folder context, meaning that for that execution, the CWD is the containing folder, this is because this way, the application.properties file would be next to the route, at least in the quick prototyping mode, offering a more flexible way to configure routes, as the opposite that would be having a single application.properties file in the workspace's root

If I understand well, the idea is to have one folder per integration project and that VS Code is opened on one parent folder having several of this "folder integration project".
If yes, this use case is currently out of scope. The amount of work for supporting completely that is too large compared of the needed work and the resources we have.

**Comment by lordrip** [comment]
Oh, no exactly like that, let me rephrase myself.

Considering the following structure:

```
my-app/
├─ kafka-consumer/
│    ├─ route.camel.yaml
│    ├─ application.properties
├─ kafka-producer/
│    ├─ route.camel.yaml
│    ├─ application.properties
├─ .gitignore
├─ application.properties
```

Considering VSCode opening `my-app/kafka-consumer/route.camel.yaml`, my proposal would be to use the `kafka-consumer` folder as CWD, this way, the `application.properties` next to that route would be picked, rather than using the root `application.properties`.

**Comment by apupier** [comment]
> Oh, no exactly like that, let me rephrase myself.
> 
> Considering the following structure:
> 
> ```
> my-app/
> ├─ kafka-consumer/
> │    ├─ route.camel.yaml
> │    ├─ application.properties
> ├─ kafka-producer/
> │    ├─ route.camel.yaml
> │    ├─ application.properties
> ├─ .gitignore
> ├─ application.properties
> ```
> 
> Considering VSCode opening `my-app/kafka-consumer/route.camel.yaml`, my proposal would be to use the `kafka-consumer` folder as CWD, this way, the `application.properties` next to that route would be picked, rather than using the root `application.properties`.

this is unfortunately the structure I meant.

**Comment by apupier** [comment]
Alternative which is covering more cases and avoids to potentially throw users in not working cases by default:
https://issues.redhat.com/browse/FUSETOOLS2-2368 Provide submenu quick action button to run with all elements
https://issues.redhat.com/browse/FUSETOOLS2-2369 Provide submenu quick action button to run with all elements from containing folder

**Comment by apupier** [comment]
closing as it won't be part of VS Code Kaoto, functional implementation is different than description and PR is available here https://github.com/camel-tooling/camel-dap-client-vscode/pull/661
